wa/workloads/gfxbench/uiauto/app/src/main/java/com/arm/wa/uiauto/gfxbench/UiAutomation.java

```diff
@@ -0,0 +1,240 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.arm.wa.uiauto.gfxbench;
+
+import android.os.Bundle;
+import android.support.test.runner.AndroidJUnit4;
+import android.support.test.uiautomator.UiObject;
+import android.support.test.uiautomator.UiObjectNotFoundException;
+import android.support.test.uiautomator.UiSelector;
+import android.support.test.uiautomator.UiScrollable;
+import android.util.Log;
+
+import com.arm.wa.uiauto.BaseUiAutomation;
+import com.arm.wa.uiauto.ActionLogger;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.concurrent.TimeUnit;
+
+@RunWith(AndroidJUnit4.class)
+public class UiAutomation extends BaseUiAutomation {
+
+    private int networkTimeoutSecs = 30;
+    private long networkTimeout =  TimeUnit.SECONDS.toMillis(networkTimeoutSecs);
+    public static String TAG = "UXPERF";
+
+    @Before
+    public void initialize(){
+        initialize_instrumentation();
+    }
+
+    @Test
+    public void setup() throws Exception{
+        setScreenOrientation(ScreenOrientation.NATURAL);
+        clearFirstRun();
+    }
+
+    @Test
+    public void runWorkload() throws Exception {
+        runBenchmark();
+        getScores();
+    }
+
+    @Test
+    public void teardown() throws Exception{
+        unsetScreenOrientation();
+    }
+
+    public void clearFirstRun() throws Exception {
+        UiObject accept = 
+            mDevice.findObject(new UiSelector().resourceId("android:id/button1")
+                .className("android.widget.Button"));
+        if (accept.exists()){
+            accept.click();
+            sleep(5);
+        }
+
+        UiObject sync = 
+                mDevice.findObject(new UiSelector().text("Data synchronization")
+                    .className("android.widget.TextView"));
+                if (sync.exists()){
+                    UiObject data =
+                        mDevice.findObject(new UiSelector().resourceId("android:id/button1")
+                            .className("android.widget.Button"));
+                    data.click();
+                }
+
+        UiObject home = 
+            mDevice.findObject(new UiSelector().resourceId("com.glbenchmark.glbenchmark27:id/tabbar_back")
+                .className("android.widget.LinearLayout"));
+            home.waitForExists(300000);
+    }
+
+    public void runBenchmark() throws Exception {
+        UiScrollable list = new UiScrollable(new UiSelector().scrollable(true));
+
+        //Disable high level tests
+        mDevice.click(840,1200);
+        UiObject highlevel = 
```

Comment:
Should factor out repetitive code  above into separate disable/enable functions.